web/controllers/screeningController/index.ts

```diff
@@ -347,6 +374,67 @@ export async function updateCourse(req: Request, res: Response) {
     }
 }
 
+async function handleNewLectures(lectures: { subcourse: { id: number }, start: Date, duration: number, instructor: { id: number } }[], courseId: number) {
+    const entityManager = getManager();
+
+    for (let i=0; i<lectures.length; i++){
+        const course = await entityManager.findOne(Course, { id: courseId });
+        if (course == undefined) {
+            logger.warn(`No course found with ID ${courseId}`);
+            return 404;
+        }
+        const subcourse = await entityManager.findOne(Subcourse, { id: lectures[i].subcourse.id, course: course });
+        if (subcourse == undefined) {
+            logger.warn(`No subcourse found with ID ${lectures[i].subcourse.id}`);
+            return 404;
+        }
+        const instructor = await entityManager.findOne(Student, { id: lectures[i].instructor.id });
+        if (instructor == undefined) {
+            logger.warn(`No instructor with ID ${lectures[i].instructor.id}`);
+            return 404;
+        }
+        if (subcourse.instructors.findIndex(i => i.id === instructor.id) == -1){
+            logger.warn(`Instructor with ID ${lectures[i].instructor.id} is not in subcourse ${subcourse.id}`);
+            return 403;
+        }
+
+        const lecture = new Lecture();
```

Comment:
Maybe we can add a constructor to Lecture, like `new Lecture(lecture, subcourse)` or `subcourse.addLecture(lecture)`?

Reply:
I added a method addLecture to Subcourse which seemed more intuitive to me.